ipset_linux.go

```diff
@@ -249,6 +254,20 @@ func (h *Handle) IpsetDel(setname string, entry *IPSetEntry) error {
 	return h.ipsetAddDel(nl.IPSET_CMD_DEL, setname, entry)
 }
 
+func encodeIP(ip net.IP) (*nl.RtAttr, error) {
+	typ := int(nl.NLA_F_NET_BYTEORDER)
+	switch len(ip) {
+	case 16:
```

Comment:
A `net.IP` variable populated via `net.ParseIP()` will be 16 bytes long even if it is parsing a IPv4 in string form.
This check may be weak.

Reply:
One better option is to test the o/p of `func (ip IP) To4() IP`

Reply:
Indeed, and the reason that it works right now is because the tests have explicit `.To4()` calls all over. A nice side effect of this will be that entries containing IPv4 addresses that are not explicitly normalized will now work.

I updated the logic there, and removed the `.To4()` calls in the tests.

Reply:
@aboch Tests are green, back to you :)